### DISCUSSION TITLE
Dynamic hydro storage carbon emissions for Portugal (and other countries?)

### DISCUSSION THREAD

**Discussion by RGarrido03** [electricitymaps/electricitymaps-contrib#4914] [Ideas] [answered]
## Description
Hi everyone! Congratulations for such an amazing app, I check it a few times a day so that I can lower my electricity-related carbon emissions.

The Portuguese grid company stores electricity during the day through hydro storage, and uses it when large amounts of energy are needed (e.g. during dinner time). I guess this is the normal behavior of a hydro storage electricity, i.e., this isn't Portugal-exclusive, right?

By default, hydro storage for Portugal has a carbon intensity of 155 gCO₂eq/kWh.

Since last week, there is a lot of wind and water in Portugal, which makes carbon emissions go as low as 40-60 gCO₂eq/kWh. That being said, the electricity stored has a carbon intensity of 40-60 gCO₂eq/kWh and not 155 gCO₂eq/kWh. However, when the electricity grid includes hydro-stored electricity, the general Portuguese carbon emissions wrongly rise up, because that portion of energy apparently has a 155 gCO₂eq/kWh carbon intensity.

I'd like to ask for dynamic hydro storage carbon intensity, which is automatically updated when electricity is stored during the day.

Thank you very much.
Rúben Garrido

**Comment by VIKTORVAV99** [comment, marked as answer]
This has been discussed before and I believe they are part of our long term goals/improvements but right now the team is very busy.

Since this would apply to both hydro storage, batteries, and other storage solutions we need to create a 2 part modell, part 1 would involve calculating the (rolling) average co2 intensity during the charge cycles (similar to how we do the flowtracing but in time rather than space). Part 2 would be to create a static co2 intensity for each type of storage taking efficiency and infrastructure into account.

As this is not a issue per say and more of a potential improvement I will move this to a discussion instead so it's easier for more people to give ideas and keep issues to problems we can "fix".
